### PR TITLE
PHP-67: Add support for custom idempotency keys

### PR DESCRIPTION
**Pull request by lighe**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1041,6 +1042,44 @@ if ($accountIdentifier instanceof BbanDetailsInterface) {
...
 
+<a name="idempotency"></a>
+
+# Custom idempotency keys
+
+By default, the client will generate and manage idempotency keys for you. However, there are cases when you might want
+to set your own idempotency keys and you can do this by using the `requestOptions` setter when creating a resource.
+
+```php
+// Create a RequestionOptionsInterface instance and set your custom idempotency key
```
Comment (lboynton) [inline README.md:1053]:
```suggestion
// Create a RequestOptionsInterface instance and set your custom idempotency key
```

File: tests/integration/ApiClient/IdempotencyKeyTest.php [hunk 2]
```diff
@@ -52,3 +52,53 @@
...
+\it('uses custom idempotency key when provided', function () {
+    $requestOptions = \client()
+        ->requestOptions()
+        ->idempotencyKey('test-idempotency-key');
+
+    \request()
+        ->requestOptions($requestOptions)
+        ->post();
+
+    expect(getRequestIdempotencyKey(1))->toBe('test-idempotency-key');
```
Comment (lboynton) [inline tests/integration/ApiClient/IdempotencyKeyTest.php:65]: Is it worth also asserting that there is only one idempotency header on the request?

**Review by lboynton — DISMISSED** [review]

**Review by lboynton — APPROVED** [review]

**Comment by stefandanaita** [comment]
Do we need a length limit or content validation for the idempotency key?

**Review by stefandanaita — APPROVED** [review]

**Comment by lighe** [comment]
> Do we need a length limit or content validation for the idempotency key?

I think we should start relying more on the API itself rather than us imposing restrictions. If the request is rejected, we will surface that as an error already.

**Pull request merged**